dataset.py

```diff
@@ -453,3 +454,75 @@ def create_fraudulent_samples(real_samples_data: pd.DataFrame, mean_isoscapes: l
     count += 1
 
   return fake_sample
+
+def valid_in_all_rasters(
+  lat: float,
+  lon: float,
+  rasters: List[raster.AmazonGeoTiff]) -> bool:
+  '''
+    Args:
+        lat (float): The latitude coordinate.
+        lon (float): The longitude coordinate.
+        rasters (List[AmazonGeoTiff]): lat, lon are checked to be valid coords
+          in each of these rasters
+
+    Returns:
+        bool: True if (lat, lon) is valid in all rasters, False if invalid in
+          at least one.
+  '''
+  for r in rasters:
+    if not raster.is_valid_point(lat, lon, r):
+      return False
+  return True
+
+def nudge_invalid_coords(df: pd.DataFrame, rasters: List[raster.AmazonGeoTiff]):
+  '''
+    Given a Pandas DataFrame with latitude and longitude columns, maybe 
+    perturb the latitude and longitude (i.e. nudge) values to fit within the 
+    bounds of every AmazonGeoTiff in `rasters`. 
+
+    This may be necessary as some rasters have slightly different coordinate 
+    systems than the one used by data providers. Samples very close to borders 
+    are particularly susceptible to being out-of-bounds and will need nudging.
+
+    Args:
+        df (pd.DataFrame): A dataframe with "lat" and "long" columns.
+        rasters (List[AmazonGeoTiff]): A list of rasters to use to decide whether
+          to nudge coordindates. If a row in the dataframe does not have a value
+          in this raster, nudge it.
+
+    Returns:
+        bool: True if (lat, lon) is valid in all rasters, False if invalid in
```

Comment:
Please change this to:
pd.DataFrame: Returns a dataframe with nudged coordinates.